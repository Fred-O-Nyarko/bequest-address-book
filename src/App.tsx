import { Box, Container, debounce, Fab } from "@mui/material";
import axios from "axios";
import React, { useCallback } from "react";
import {
  AddressModal,
  EmptyState,
  FloatingActionButton,
  Loader,
  SearchBox,
} from "./components";

const App = () => {
  const [addresses, setAddresses] = React.useState([]);
  const [loading, setLoading] = React.useState(false);
  const [openModal, setOpenModal] = React.useState(false);
  const [options, setOptions] = React.useState([]);

  const fetchData = (value: string) => {
    setLoading(true);
    axios
      .get(
        `https://api.getAddress.io/find/${value}?api-key=5LOTLJcma065xnWNvF4Bbg36195`
      )
      .then((res) => {
        console.log(res.data);
        setLoading(false);
      })
      .catch((err) => {
        console.log(err);
        setLoading(false);
      });
  };

  const onInputChange = useCallback(debounce(fetchData, 1000), []);

  const getCounties = (value: React.ChangeEvent<Element>) => {
    setLoading(true);
    axios
      .post(
        `https://api.getAddress.io/typeahead/${value}?api-key=5LOTLJcma065xnWNvF4Bbg36195`,
        {
          search: ["country"],
        }
      )
      .then((res) => {
        console.log(res.data);
        setLoading(false);
        setOptions(res.data);
      })
      .catch((err) => {
        console.log(err);
        setLoading(false);
      });
  };

  console.log(addresses);
  return (
    <Container
      style={{
        padding: "4rem",
        display: "flex",
        alignItems: "center",
        flexDirection: "column",
        position: "relative",
        height: "100vh",
      }}
      maxWidth="sm"
    >
      <SearchBox
        options={addresses}
        loading={loading}
        onChange={() => console.log("here")}
        label={loading ? "Loading..." : "Search Address"}
      />
      <Box marginTop={8}>
        {loading ? (
          <Loader size="4rem" />
        ) : (
          <EmptyState message="It's kinda lonely here ðŸ˜¢" />
        )}
      </Box>
      {openModal && (
        <AddressModal
          open={openModal}
          setOpenModal={setOpenModal}
          onSearch={getCounties}
          options={options}
          loading={loading}
        />
      )}
      <FloatingActionButton onClick={() => setOpenModal(true)} />
    </Container>
  );
};

export default App;
