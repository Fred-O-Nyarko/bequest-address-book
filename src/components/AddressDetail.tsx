import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  List,
  ListItem,
  ListItemText,
  Typography,
} from "@mui/material";
import { IAddress, TModalID } from "../shared/types";

interface IAddressDetailProps {
  open: TModalID;
  setOpenModal: (m: TModalID) => void;
  addressDetail: IAddress | null;
}

const AddressDetail = ({
  open,
  setOpenModal,
  addressDetail,
}: IAddressDetailProps) => {
  return (
    <Dialog open={!!open} onClose={() => setOpenModal(null)} fullWidth>
      <DialogTitle>Address Detail</DialogTitle>
      <DialogContent dividers>
        <List>
          <ListItem>
            <ListItemText
              primary="Address Line 1"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={<Typography>{addressDetail?.lineOne}</Typography>}
            />
          </ListItem>
          <ListItem>
            <ListItemText
              primary="Address Line 2"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={<Typography>{addressDetail?.lineTwo ?? 'N/A'}</Typography>}
            />
          </ListItem>
          <ListItem>
            <ListItemText
              primary="Address Line 3"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={<Typography>{addressDetail?.lineThree ?? 'N/A'}</Typography>}
            />
          </ListItem>
          <ListItem>
            <ListItemText
              primary="Postcode"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={<Typography>{addressDetail?.postCode}</Typography>}
            />
          </ListItem>
          <ListItem>
            <ListItemText
              primary="Town/City"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={<Typography>{addressDetail?.town}</Typography>}
            />
          </ListItem>
          <ListItem>
            <ListItemText
              primary="Country"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={<Typography>{addressDetail?.country}</Typography>}
            />
          </ListItem>
        </List>
      </DialogContent>
      <DialogActions>
        <Button onClick={() => setOpenModal(null)}>Cancel</Button>
      </DialogActions>
    </Dialog>
  );
};

export default AddressDetail;
