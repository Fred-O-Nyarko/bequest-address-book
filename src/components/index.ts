export {default as SearchBox} from './SearchBox';
export {default as EmptyState} from './EmptyState'
export {default as Loader} from './Loader'
export {default as FloatingActionButton} from './FloatingActionButton'
export {default as AddressModal} from './AddressModal'
export {default as AddressListItem} from './AddressListItem'
export {default as Notification } from './Notification'