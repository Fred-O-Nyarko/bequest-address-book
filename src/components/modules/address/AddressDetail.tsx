import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  List,
  ListItem,
  ListItemText,
  Typography,
} from "@mui/material";
import {
  useAppDispatch,
  useAppSelector,
  selectModal,
  setModal,
  selectAddress,
} from "src/redux";


const AddressDetail = () => {
  const dispatch = useAppDispatch();
  const modal = useAppSelector(selectModal);
  const addressDetail = useAppSelector(selectAddress);
  
  const closeModal = () => {
    dispatch(setModal(null));
  };

  return (
    <Dialog open={!!modal} onClose={closeModal} fullWidth>
      <DialogTitle>Address Detail</DialogTitle>
      <DialogContent dividers>
        <List>
          <ListItem>
            <ListItemText
              primary="Address Line 1"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={<Typography>{addressDetail?.lineOne}</Typography>}
            />
          </ListItem>
          <ListItem>
            <ListItemText
              primary="Address Line 2"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={
                <Typography>{addressDetail?.lineTwo ?? "N/A"}</Typography>
              }
            />
          </ListItem>
          <ListItem>
            <ListItemText
              primary="Address Line 3"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={
                <Typography>{addressDetail?.lineThree ?? "N/A"}</Typography>
              }
            />
          </ListItem>
          <ListItem>
            <ListItemText
              primary="Postcode"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={<Typography>{addressDetail?.postCode}</Typography>}
            />
          </ListItem>
          <ListItem>
            <ListItemText
              primary="Town/City"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={<Typography>{addressDetail?.town}</Typography>}
            />
          </ListItem>
          <ListItem>
            <ListItemText
              primary="Country"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={<Typography>{addressDetail?.country}</Typography>}
            />
          </ListItem>
        </List>
      </DialogContent>
      <DialogActions>
        <Button onClick={closeModal}>Close</Button>
      </DialogActions>
    </Dialog>
  );
};

export default AddressDetail;
