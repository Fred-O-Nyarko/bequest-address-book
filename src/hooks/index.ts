export { useAddressForm} from './useAddressForm';
export { useDebounce } from './useDebounce';