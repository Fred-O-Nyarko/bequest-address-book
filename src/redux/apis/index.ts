export * from "./countries";
export * from "./addresses";
