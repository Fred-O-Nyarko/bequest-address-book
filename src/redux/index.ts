export { default as store } from "./store";
export * from "./middlewares";
export * from "./hooks";
export * from "./slices";