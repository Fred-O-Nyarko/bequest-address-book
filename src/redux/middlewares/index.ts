export { default as errorMiddleware } from './apiError';