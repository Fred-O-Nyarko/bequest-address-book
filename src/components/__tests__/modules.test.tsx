/* eslint-disable testing-library/prefer-screen-queries */
import { cleanup, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { setModal, store } from "src/redux";
import AddressDetail from "../modules/address/AddressDetail";

describe("components test for modules", () => {
  afterEach(cleanup);
  it("renders AddressDetail component", () => {
    store.dispatch(setModal("detail-form"));

    const { getByText, container } = render(
      <Provider store={store}>
        <AddressDetail />
      </Provider>
    );

    expect(getByText(/Address Detail/)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
