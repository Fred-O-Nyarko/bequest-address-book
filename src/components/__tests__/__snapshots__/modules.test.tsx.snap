// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components test for modules renders AddressDetail component 1`] = `
<div
  aria-hidden="true"
/>
`;
